src/main/webapp/user_profile_script.js

```diff
@@ -12,17 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function deleteRow(row) {
-    if (confirm("Are you sure that you want to delete this already published task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("need-help").deleteRow(rowIndex);
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info);
+    return info;
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
+        window.alert("You can only delete an 'OPEN' task.")
+    } else {
+        if (confirm("Are you sure that you want to delete the task?")) {
+            const queryURL = "/tasks?key=" + keyString;
+            const request = new Request(queryURL, {method: "DELETE"});
+            const response = await fetch(request);
+        }
+    }
+}
+
+async function editTask(keyString) {
+    const info = await getInfo(keyString);
+    console.log(info);
+    if (info.status !== "OPEN") {
+        window.alert("You can only edit an 'OPEN' task.")
+    } else {
+        document.getElementById("edit-content-input").value = info.detail;
+        document.getElementById("edit-point-input").value = info.reward.toString();
+        const id_input = document.getElementById("task-id-input");
+        id_input.value = info.keyString;
+        id_input.readOnly = true;
+        id_input.style.display = "none"
+        document.getElementById("editTaskModal").style.display = "block";
     }
 }
 
-function completeTask(row) {
-    if (confirm("Are you sure that you have already completed the task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("offer-help").deleteRow(rowIndex);
+async function completeTask(keyString) {
```

Comment:
I'm not sure how to create a task locally as one user and look at it as another user. As a result, I don't know how to get my local instance into a state where I can try assigning a task to myself and marking it complete. Do you have any advice for how to test this?

Reply:
Actually I didn't consider this kind of situation when I wrote the code 😭. So I don't have any good advice on how to test this. But maybe we could prevent this kind of situation by not allowing the user to assign the task to a certain people. Instead, users have to undertake tasks on the main page, and when displaying tasks on the main page, we will filter out all the tasks that are published by the current user. In this way, a user can never complete a task that is published by himself. 

Apart from this, I modify the `completeTask` function in the new commit such that before changing the status of the task, it would first check the owner and the helper of the task. If they are the same user, it would send an alert and then delete the task instead of changing the status to "COMPLETE: AWAIT VERIFICATION". I think this could be a good solution to this edge case.

Reply:
Sorry, I just meant that I can't see how to log in as a different user to try and assign some tasks to that user. But with the future PRs in this repo that I've now seen, hopefully that will be possible soon.

How have you been testing the "Offer help" side of this page, including completing tasks and such?